src/server/list_family.cc

```diff
@@ -146,23 +173,43 @@ class BPopPusher {
 
 // Called as a callback from MKBlocking after we've determined which key to pop.
 std::string OpBPop(Transaction* t, EngineShard* shard, std::string_view key, ListDir dir) {
+  DVLOG(2) << "popping from " << key << " " << t->DebugId();
+
   auto& db_slice = shard->db_slice();
   auto it_res = db_slice.Find(t->GetDbContext(), key, OBJ_LIST);
+
+  if (!it_res) {
+    auto messages = debugMessages.All();
+    cerr << "CRASH REPORT" << endl;
+    cerr << "key: " << key << " tx: " << t->DebugId() << "\n";
+    cerr << "===\n";
+    for (auto msg : messages)
```

Comment:
Nit 1: use `const auto&` (or `const string&`) to avoid copying the message
Nit 2: You could use `for (const auto& msg : debugMessages.All())` and not save the var locally